test/parallel/test-fs-rmdir-recursive.js

```diff
@@ -144,8 +151,8 @@ function removeAsync(dir) {
   // Recursive removal should succeed.
   await fs.promises.rmdir(dir, { recursive: true });
 
-  // No error should occur if recursive and the directory does not exist.
-  await fs.promises.rmdir(dir, { recursive: true });
+  // Should print deprecation warning if recursive and directory does not exist.
```

Comment:
I'm wondering if this will only call `expectWarning` once, because the warning is suppressed on the second call (_I would expect us to have a `common. expectWarning` for each instance where we expect the warning to happen.

Is there an example elsewhere in the codebase of a test that verifies multiple warnings?

Reply:
I searched the tests for `expectWarning` and was not able to find an example of this. I didn't look at every test but I checked quite a few. I agree that I would like this test to be more thorough. I'll look into it some more but am open to ideas.

Reply:
Can we move the `common.expectWarning` to live right above where we call the `method` that will warn, than perhaps we can clear the require cache between calls, so that we test each scenario should should warn?

It would be good to have a test case for the modules usages that don't warn too.

Reply:
Let's just create two additional test files fro the warning logic:

```
test/parallel/test-fs-rmdir-recursive-warns-not-found.js
test/parallel/test-fs-rmdir-recursive-warns-on-file.js
```